src/Dapr.Client/DaprClient.cs

```diff
@@ -313,6 +313,21 @@ public HttpRequestMessage CreateInvokeMethodRequest<TRequest>(string appId, stri
         /// <param name="cancellationToken">A <see cref="CancellationToken" /> that can be used to cancel the operation.</param>
         /// <returns>A <see cref="Task{T}" /> that will return the value when the operation has completed.</returns>
         public abstract Task<bool> CheckHealthAsync(CancellationToken cancellationToken = default);
+
+        /// <summary>
+        /// Perform health-check of Dapr sidecar's outbound APIs. Return 'true' if the sidecar is health. Otherwise false.
+        /// </summary>
+        /// <param name="cancellationToken">A <see cref="CancellationToken" /> that can be used to cancel the operation.</param>
+        /// <returns>A <see cref="Task{T}" /> that will return the value when the operation has completed.</returns>
+        public abstract Task<bool> CheckOutboundHealthAsync(CancellationToken cancellationToken = default);
+
+        /// <summary>
+        /// Calls <see cref="CheckHealthAsync(CancellationToken)"/> until the sidecar is reporting as healthy. This method checks once a second
+        /// and by default will check 30 times, totaling a 30 second wait.
```

Comment:
The default timeout here should just be infinite and the cancellation token required. The problem is that any default timeout will be **wrong for someone** and thus needs to be customizable. Cancellation is already the right way to signal a timeout 😆

Reply:
Yeah, this comment was misleading as we did switch to the `CancellationToken` but forgot to update the comment :)